TestTicTacToe/BoardTests.cs

```diff
@@ -0,0 +1,65 @@
+using TicTacToe;
+using Xunit;
+
+namespace TestTicTacToe
+{
+    public class BoardTests
+    {
+        [Fact]
+        public void Should_Place_Marker_In_The_Correct_Position_If_Empty_Cell()
+        {
+            // Arrange 
+            var board = new Board();
+            var move = new Move(1 - Constants.IndexingAdjustment, 1 - Constants.IndexingAdjustment);
+            
+            var expectedBoard = new[,]
+            {
+                {1, 0, 0},
+                {0, 0, 0},
+                {0, 0, 0}
+            };
+        
+            // Act
+            board.PlaceMarker(move, 1);
+            var actualBoard = board.GetBoard();
+        
+            // Assert
+            Assert.Equal(expectedBoard, actualBoard);
+        }
+
+        [Fact]
+        public void Should_Reset_The_Board_Back_To_Its_Initial_State()
+        {
+            // Arrange 
+            var board = new Board();
+            var firstMove = new Move(1 - Constants.IndexingAdjustment, 1 - Constants.IndexingAdjustment);
+            var secondMove = new Move(1 - Constants.IndexingAdjustment, 2 - Constants.IndexingAdjustment);
+            
+            var expectedBoardBeforeReset = new[,]
+            {
+                {1, 2, 0},
+                {0, 0, 0},
+                {0, 0, 0}
+            };
+            
+            var expectedBoardAfterReset = new[,]
+            {
+                {0, 0, 0},
+                {0, 0, 0},
+                {0, 0, 0}
+            };
+
+            // Act
+            board.PlaceMarker(firstMove, 1);    
+            board.PlaceMarker(secondMove, 2);    
+            var actualBoardBeforeReset = board.GetBoard();
+            board.ResetBoard();
+            var actualBoardAfterReset = board.GetBoard();
+        
+            // Assert
+            Assert.Equal(expectedBoardBeforeReset, actualBoardBeforeReset);
```

Comment:
Be careful with having tests with multiple asserts, sometimes they can add confusion when you are trying to find why a test is failing. It also adds more things in your test that you have to maintain. You have tests that cover the PlaceMarker so it should be pretty safe to just assert that the board returns to empty.
I think in terms of revealing intent I would look at changing the name of the test from Should_Reset_The_Board_Back_To_Its_Initial_State to something like Should_Reset_The_Board_Back_To_Empty/Start, to me initial state could mean there are some starting values in there already, rather than it being all 0's.